### PR TITLE
Refactor FXIOS-8361 Update mobile icons to new style pt4

### PR DESCRIPTION
**Pull request by thatswinnie**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8361)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18509)

## :bulb: Description
It's been a long time coming: This is the last PR for our icon updates! Because of these updates we were able to remove 5 asset catalogs in total!
The PR updates:
- Cloud icon
- Clear icon
- Purple private mode icon
- Tabs icon in toolbar & tab tray
- Device icons (desktop, mobile, tablet)
- Brightness icon in reader mode
- Sponsored star icon
- Fakespot stars icons
- Minus icon in zoom feature
- Lightning icon in QR code reader

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Coordinators/ShareExtensionCoordinator.swift [hunk 1]
```diff
@@ -98,7 +98,7 @@ class ShareExtensionCoordinator: BaseCoordinator,
         let themeColors = themeManager.currentTheme(for: windowUUID).colors
         let colors = SendToDeviceHelper.Colors(defaultBackground: themeColors.layer1,
                                                textColor: themeColors.textPrimary,
-                                               iconColor: themeColors.iconPrimary)
+                                               iconColor: themeColors.iconDisabled)
```
Comment (dataports) [inline firefox-ios/Client/Coordinators/ShareExtensionCoordinator.swift:101]: just checking - using the disabled color here was intentional yes?

File: firefox-ios/Client/Frontend/Browser/Tabs/Legacy/LegacyTabTrayViewController.swift [hunk 2]
```diff
@@ -529,7 +529,7 @@ extension LegacyTabTrayViewController: Notifiable {
                 else { return }
                 self?.countLabel.text = self?.viewModel.normalTabsCount
                 self?.segmentedControlIphone.setImage(
-                    UIImage(named: ImageIdentifiers.navTabCounter)!.overlayWith(image: label),
+                    UIImage(named: StandardImageIdentifiers.Large.tab)!.overlayWith(image: label),
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Browser/Tabs/Legacy/LegacyTabTrayViewController.swift:532]: Should we be force unwrapping the image here?

**Review by dataports — APPROVED** [review]
A couple of non-critical comments or questions but looks great otherwise!

File: firefox-ios/Client/Coordinators/ShareExtensionCoordinator.swift (hunk 1 above)
Comment (thatswinnie) [reply firefox-ios/Client/Coordinators/ShareExtensionCoordinator.swift:101]: Yes, it's intentional. It was discussed with @cwzilla

**Review by isabelrios — APPROVED** [review]

**Pull request merged**